api/api/configuration.py

```diff
@@ -0,0 +1,62 @@
+# Copyright (C) 2015-2019, Wazuh Inc.
+# Created by Wazuh, Inc. <info@wazuh.com>.
+# This program is a free software; you can redistribute it and/or modify it under the terms of GPLv2
+from wazuh import common
+import os
+import yaml
+
+
+def fill_dict(default, config):

```

Comment:
I would suggest a different approach. Why don't we take the default config dict and update it using the user configuration? A pythonic way to do this could be:
```python
default_configuration = {'param1': 'value1', 'param2: 'value2'}
user_configuration = {'param2': 'user_value2'}
configuration = {**default_configuration, **user_configuration}
```

Reply:
That suggestion looks nice but it has a tiny problem: it won't merge nested dictionaries:
```python
>>> default_configuration = {'param1': 'value1', 'param2': {'p1': 'v1', 'p2': 'v2'}}
>>> user_configuration = {'param2': {'p1': 'v3'}}
>>> {**default_configuration, **user_configuration}
{'param1': 'value1', 'param2': {'p1': 'v3'}}
```